### ISSUE TITLE
Day 3 Review: Documentation

### ISSUE THREAD

**Issue by belindathemoon** [semaphoreP/codeastro#117]
### How clear is the material? What’s confusing?
Overall pretty clear and I like the screen shots. The only potential pitfalls I found is that if someone has subdirectories, they may get confused and think they need to put the docs directory in each subdirectory, so make clear that it just goes in the top level of your package.

### Are we assuming any knowledge that we shouldn’t be? Where should we add background info?
Sphinx asks project release. Background would be useful here on how to name the release, as well as the difference between release and version. 

### Is the structure of the lesson engaging? Are there any parts that are unnecessary/boring/too long?
I thought it was great, but I am also a fan of documentation. :) If anything, I think there could be a _little_ more time spent on the Google-style docstrings.

### how much time did it take you to complete the assignments?
I think they'll need more than 25 minutes. It's a simple activity in some ways, but it's also completely new, so they'll be referring back to the page a lot. It took me about 25 minutes and I've done this before.

**Comment by Palma-Bifani** [comment]
- How clear is the material? What’s confusing?

I like it a lot. Maybe the twine part could be presented in more detail. When I was organizing a similar workshop for the Chilean community, we made a jupyter notebook explaining some of the files and the install process in more detail (see link below)

[https://github.com/Science-and-Commit/Workshop_2022/blob/main/4_code_profilling_and_releasing/releasing_code.ipynb](https://github.com/Science-and-Commit/Workshop_2022/blob/main/4_code_profilling_and_releasing/releasing_code.ipynb)

- Are we assuming any knowledge that we shouldn’t be? Where should we add background info?

The only thing I can think about is that they may not be ready with the package and therefore tey will not be able to complete this task in the given activity time... but they can complete it afterwards.

- Is the structure of the lesson engaging? Are there any parts that are unnecessary/boring/too long?

I think this is one of the most expected lessons of the workshop! They will be very interested.

- How much time did it take you to complete the assignments?

I agree that completing this task may take around 30 min.

**Comment by Palma-Bifani** [comment]
- How clear is the material? What’s confusing?

I think that the name of the file is not clear on day3. It should be probably something like "Documentation Example" rather than "exampledoc".

IIs it possible that the screen shots are imported as code sections? Would be easier to copy-paste while following the activity...


- Are we assuming any knowledge that we shouldn’t be? Where should we add background info?

It could be worth showing an example of a nice code documented with Sphinx in order for the participants to know what they should aim for. Probably not all of them know. And comparing Sphinx to ReadTheDocs could be nice too. I mean, maybe is worth giving more background on what is expected from a good documentation setup.


- Is the structure of the lesson engaging? Are there any parts that are unnecessary/boring/too long?

I think that this is not the most engaging lesson but it is nicely done and they will appreciate having it when they will be doing the activity. The Sphinx quickstart guide is great. 


- How much time did it take you to complete the assignments?

I think this is highly dependent on their status in the activity. But if things run smoothly, probably they will need 30 min...

**Comment by mwhosek** [comment]
Great, thanks so much for the feedback @belindathemoon and @Palma-Bifani! I've merged updates to the documentation into main that hopefully addresses your comments (done 6/20). Specifically:

@belindathemoon:
-Good point regarding the "docs" sub-directories. I'll be sure to mention that verbally as I go through the lesson
-We are going to do the lesson for releasing code right before this one, so folks will have more background regarding code releases/version numbers :).
-Unfortunately our timing is a bit tight with everything else we are doing on Day 3, but I'm going to try to give folks 30 mins to work on this themselves if possible. However, I suspect that many will end up needing to continue working on their code documentation into the afternoon session.

@Palma-Bifani 
-I've changed the name of the directory to "tutorial_doc", which hopefully is more clear. I didn't want to have the directory name start with "doc_", in fear it would be confusing with the "docs" subdirectory where folks compile sphinx.
-Regarding the screenshots: Many of the screenshots involve making slight edits to files that sphinx automatically creates, and so I wanted to use screenshots to emphasize where those edits needed to be made. I also felt that having participants make these changes manually would be helpful to follow what is happening. However, when the participants need to make the `correlate.rst` file from scratch in step 9, I included a code section for folks to copy-paste in addition to a screen-shot describing the file. Hopefully this balances the convenience of copy-paste with participants following the edits themselves :). 
-Great idea to include more information regarding the additional aspects of documenting code as well as including examples for reference! I've added an "Activity Wrap-Up" section at the end, which I'll go through in the last 5 minutes before we finish the activity. Unfortunately we don't have much time to go beyond the basics of docstrings + sphinx in this particular lesson, but I hope this wrap-up will provide a useful starting point folks continue to work on their projects.
-Regarding ReadTheDocs: Unfortunately we are already tight on time and so I don't think we will be able to go into more detail there. I've included a bonus afternoon activity that involves participants posting their sphinx documentation to ReadTheDocs, but I consider this more of a "stretch goal" for folks who are already familiar with code documentation.

Thanks again for your excellent feedback! Let me know if you have any further comments regarding the lesson.
